display/core.py

```diff
@@ -49,54 +50,133 @@ def save(self, outdir, filetype='png'):
-                 dx=600, dy=600):
+                 dx=600, dy=600, pads=[""]):
         self.projection = projection
-        tx = 50 + self.__class__.nviews * (dx+10) 
+        self.hists = dict()
+        self.registered = dict()
+        self.locked = dict()
+        tx = 50 + self.__class__.nviews * (dx+10)
         ty = 50
-        self.canvas = TCanvas(name, name, tx, ty, dx, dy)
+        width = 1
+        height = 1
+        self.npads = 1
+        self.pads = dict()
+ 
+        if not pads is None:
+            self.npads = len(pads)
+            if self.npads%8 == 0 and self.npads >= 8:
+                width = width*4
+                height = height *self.npads/4
+                dx = dx/2
+                dy = dy/2
+                xmax = 2
+                ymin = -xmax
+                xmin = -xmax
+                ymax = xmax
+
+            else:
+                width = width * self.npads
+        else:
+            pads = [""]
+            
+        self.canvas = TCanvas(name, name, tx, ty, width *dx, height*dy)
+        self.canvas.SetRightMargin(0.)
+        self.canvas.SetLeftMargin(0.)
+        self.canvas.SetTopMargin(0.) 
+        self.canvas.SetBottomMargin(0.) 
+                   
+        self.canvas.Divide(width, height)
+            
+        for x in range(0, self.npads):
+            c1 = self.canvas.cd(x+1)
+            c1.SetLeftMargin(0.0015)
+            c1.SetRightMargin(0.0015)  
+            c1.SetTopMargin(0.0015)  
+            c1.SetBottomMargin(0.0015)  
+            panename = name + ": " + pads[x]
+            self.pads[x] = ViewPad(panename, name, nx, xmin, xmax, ny, ymin, ymax, side=x)
+            
+        self.__class__.nviews += 1 
+        
+    def register(self, obj, layer, clearable=True, sides=None):
+        if sides is None:
+            sides = range(0, len(self.pads))
+        
+        for x in sides:
+            self.pads[x].register(obj, layer, clearable)
+
+    def clear(self):
+        for p in self.pads.itervalues():
+            p.clear()
+        
+    def draw(self):
+        for x in range(0, self.npads):
+            if self.canvas is None:
+                pass
+            else:
+                self.canvas.cd(x+1)
+                self.pads[x].draw()
+                self.canvas.Update()
+
+    def zoom(self, xmin, xmax, ymin, ymax):
+        for p in self.pads.itervalues():
+            p.zoom(xmin, xmax, ymin, ymax)
+        self.draw()    
+
+    def unzoom(self):
+        for p in self.pads.itervalues():
+            p.unzoom()
+            
+        #self.canvas.Modified()
+        self.draw()   
+
+    def save(self, outdir, filetype):
+        fname = '{outdir}/{name}.{filetype}'.format(outdir=outdir,
+                                                    name=self.canvas.GetName(),
+                                                    filetype=filetype)
+        self.canvas.SaveAs(fname)
+
+  
+class ViewPad(object):
```

Comment:
I don't think it is worthwhile. They have the same method names but ViewPane says "call do something on each ViewPad". View pad says "do something"